### ISSUE TITLE
FileCache.php bug at line 140 in protected function getFilename

### ISSUE THREAD

**Issue by mtal** [doctrine/cache#121]
(($this->isRunningOnWindows && $this->directoryStringLength + 4 + strlen($id) * 2 + $this->extensionStringLength) > 259) is always false
BECAUSE
($this->isRunningOnWindows && $this->directoryStringLength + 4 + strlen($id) * 2 + $this->extensionStringLength) is boolean
AND 
259 is too high for windows which cause "fopen failed to open stream: Invalid argument doctrine cache"

Original
```PHP
    protected function getFilename($id)
    {
        $hash = hash('sha256', $id);

        // This ensures that the filename is unique and that there are no invalid chars in it.
        if (
            '' === $id
            || ((strlen($id) * 2 + $this->extensionStringLength) > 255)
            || (($this->isRunningOnWindows && $this->directoryStringLength + 4 + strlen($id) * 2 + $this->extensionStringLength) > 259)
        ) {
            // Most filesystems have a limit of 255 chars for each path component. On Windows the the whole path is limited
            // to 260 chars (including terminating null char). Using long UNC ("\\?\" prefix) does not work with the PHP API.
            // So if the id in hex representation would surpass the limit, we use the hash instead. The prefix prevents
            // collisions between the hash and bin2hex.
            $filename = '_' . $hash;
        } else {
            $filename = bin2hex($id);
        }
```

**Comment by zeroedin-bill** [comment]
Should be resolved in master and 1.5.3

**Comment by zeroedin-bill** [comment]
refs #122
